casbin/enforcer.cpp

```diff
@@ -173,8 +173,8 @@ Enforcer* Enforcer :: NewEnforcer() {
  * @param model_path the path of the model file.
  * @param policyFile the path of the policy file.
  */
-Enforcer* Enforcer :: NewEnforcer(string model_path, string policyFile) {
-    return NewEnforcer(model_path, FileAdapter :: NewAdapter(policyFile));
+shared_ptr<Enforcer> Enforcer :: NewEnforcer(string model_path, string policyFile) {
+    return NewEnforcer(model_path, shared_ptr<FileAdapter>(FileAdapter :: NewAdapter(policyFile)));
```

Comment:
Really need to pass in a `shared_ptr<FileAdapter>`? I think the adapter will not be referenced anywhere